lncfg/db.go

```diff
@@ -78,56 +79,189 @@ func (db *DB) Init(ctx context.Context, dbPath string) error {
 	return nil
 }
 
+// boltBackendCreator is a type alias for a function that returns a bolt DB
+// backend with the given bolt specific configuration.
+type boltBackendCreator func(boltCfg *kvdb.BoltConfig) (kvdb.Backend, error)
+
 // DatabaseBackends is a two-tuple that holds the set of active database
-// backends for the daemon. The two backends we expose are the local database
-// backend, and the remote backend. The LocalDB attribute will always be
-// populated. However, the remote DB will only be set if a replicated database
-// is active.
+// backends for the daemon. The two backends we expose are the graph database
+// backend, and the channel state backend.
 type DatabaseBackends struct {
-	// LocalDB points to the local non-replicated backend.
-	LocalDB kvdb.Backend
+	// GraphDB points to the database backend that contains the less
+	// critical data that is accessed often, such as the channel graph and
+	// chain height hints.
+	GraphDB kvdb.Backend
```

Comment:
Why the split between graph and channel? For bbolt that data is stored in the same file and only a single instance was used (somewhere it says `remoteChanDB = localChanDB`). By making it two instances, the global lock semantics change and it would need to be checked that nowhere this logic is relied upon.

Reply:
We did a scan when we did the initial split for the current remote graph portion, and we didn't find any cases where locking between the graph and the channel graph was critical for consistency. Even now we have several databases (watch tower, macaroons, wallet, etc), and if we need to we ensure consistency on restart by applying certain operations again (trying to rebroadcast a transaction as an example), etc. 

Also conceptually, the graph database and the channel database never are linked during routine operations. The only intersection points I can think up off the top are: path finding, and then applying a channel update due to a failed payment. Neither of which need to happen in lock step.

Reply:
But why the split between these two databases now? We come from bbolt where there was just a single instance serving both data sets. I don't mind, but it seems unnecessary.

Reply:
On the other hand, splitting them up may reduce lock contention on the global write lock. But still it looks to me like an independent change.

Reply:
I feel like _if_ we're ever going to split them, then _now_ is the time. Adding the namespace to the channel state and wallet DBs will mean that existing `etcd` deployments won't be compatible with `v0.14.0` anyway (which might be another discussion entirely, do we need backward compatibility to `etcd` clusters that were spun up with `v0.13.x`? since it's clearly marked as experimental I hope nobody just migrated their mainnet node?).

Reply:
If you indeed decide to split now, you will also need to look at the migration logic. Currently that all runs on just the combined database and will need to be split as well?

Reply:
The schema migration is definitely a good argument. I decided to not do the split in this PR. But I'll start a new PR on top of this one where I look into how much work would need to go into this.